src/features/unified-share-modal/ContactRestrictionNotice.js

```diff
@@ -0,0 +1,139 @@
+    restrictedExternalEmails: Array<string>,
+    selectedContacts: Array<Contact>,
+    selectedJustificationReason: ?SelectOptionProp,
+} & InjectIntlProvidedProps;
+
+const ContactRestrictionNotice = ({
+    error,
+    intl,
+    isLoading,
+    justificationReasons,
+    onRemoveRestrictedExternalContacts,
+    onSelectJustificationReason,
+    restrictedExternalEmails,
+    selectedContacts,
+    selectedJustificationReason,
+}: Props) => {
+    const compMessageId = uniqueId('compMessage');
+    const restrictedExternalContacts = selectedContacts.filter(({ value }) => restrictedExternalEmails.includes(value));
+    const restrictedExternalContactCount = restrictedExternalContacts.length;
+
+    if (!restrictedExternalContactCount) {
+        return null;
+    }
+
+    const RemoveButton = ({ children }: { children: React.Node }) => (
+        <PlainButton
+            className="bdl-ContactRestrictionNotice-removeBtn"
+            data-resin-target="removeBtn"
+            onClick={onRemoveRestrictedExternalContacts}
+        >
+            {children}
+        </PlainButton>
+    );
+
+    // TODO:
+    // Switch to react-intl v3+ api once migration is complete. FormattedCompMessage is now
+    // deprecated and has some issues with components nested within <Plural/>, hence why we
+    // use two messages and use unique keys as a workaround.
+
+    const noticeDescriptionSingular = (
+        <FormattedCompMessage
+            key={compMessageId}
+            description="Notice to display when sharing a file with external collaborators requires a business justification to be provided."
+            id="boxui.unifiedShare.businessJustificationRequiredSingular"
+        >
+            This classified content requires business justification to collaborate with{' '}
+            <ContactsEmailsTooltip contacts={restrictedExternalContacts}>1 person</ContactsEmailsTooltip>. Select a
+            business justification below or <RemoveButton>remove them</RemoveButton> to continue.
+        </FormattedCompMessage>
+    );
+
+    const noticeDescriptionPlural = (
+        <FormattedCompMessage
+            key={compMessageId}
+            description="Notice to display when sharing a file with external collaborators requires a business justification to be provided."
+            id="boxui.unifiedShare.businessJustificationRequiredPlural"
+        >
+            This classified content requires business justification to collaborate with{' '}
+            <ContactsEmailsTooltip contacts={restrictedExternalContacts}>
+                <Param
+                    value={restrictedExternalContactCount}
+                    description="Number of external collborators currently selected"
+                />{' '}
+                people
+            </ContactsEmailsTooltip>
+            . Select a business justification below or <RemoveButton>remove them</RemoveButton> to continue.
```

Comment:
Yes, so in this case we have the two separate versions of the message, so I think it shouldn't be an issue. The problem I think is that we're not currently using a different copy for "remove them", but I think "remove them" also works when referring to one person?

The `{' '}` are being automatically added by the linter and I think would probably be added when committing the code even if I forced them to be removed. I think translators would be working mostly with the properties files, but let me double-check with moji.

Reply:
But yes, let me confirm the "requires a business justification" part

Reply:
I'd check with the internationalization team on this. In English, we often have 2-3 variants for plurals (_none_, _one_ or _many_), but in other languages, it could more depending on how the verbs conjugate. That's why we use the intl framework pluralization methods.

Reply:
Yes, confirmed that this workaround might cause issues in some languages. Currently looking into a fix for `<FormattedCompMessage />` that might get `<Plural />` working properly.